azure-communication-ui/azure-communication-ui-demo-app/src/main/java/com/azure/android/communication/ui/callingcompositedemoapp/SettingsActivity.kt

```diff
@@ -0,0 +1,235 @@
+        }
+    }
+
+    private fun initializeViews() {
+        callSettingLabelView = findViewById(R.id.callSellingLabelView)
+        languageSettingLabelView = findViewById(R.id.languageSettingLabelView)
+        languageSettingLabelDivider = findViewById(R.id.languageSettingLabelDivider)
+        isRTLCheckBox = findViewById(R.id.languageIsRTL)
+        languageAdapterLayout = findViewById(R.id.languageAdapterLayout)
+        autoCompleteTextView = findViewById(R.id.autoCompleteTextView)
+        customTranslationCheckBox = findViewById(R.id.customTranslation)
+    }
+
+    private fun languageSettingsExperience() {
+        if (!SettingsFeatures.isLanguageFeatureEnabled) {
+            languageSettingLabelView.visibility = View.GONE
+            languageSettingLabelDivider.visibility = View.GONE
+            languageAdapterLayout.visibility = View.GONE
+            isRTLCheckBox.visibility = View.GONE
+            customTranslationCheckBox.visibility = View.GONE
+        } else {
+            languageSettingLabelView.visibility = View.VISIBLE
+            languageSettingLabelDivider.visibility = View.VISIBLE
+            languageAdapterLayout.visibility = View.VISIBLE
+            isRTLCheckBox.visibility = View.VISIBLE
+            customTranslationCheckBox.visibility = View.VISIBLE
+        }
+        SettingsFeatures.setIsLanguageFeatureEnabled(
+            this,
+            SettingsFeatures.isLanguageFeatureEnabled
+        )
+    }
+
+    private fun updateRTLCheckbox() {
+        val selectedLanguage = getSelectedLanguageValue()
+        val isRTLKey = LANGUAGE_ISRTL_VALUE_SHARED_PREF_KEY + selectedLanguage
+        if (selectedLanguage != null) {
+            isRTLCheckBox.isChecked =
+                sharedPreference.getBoolean(isRTLKey, DEFAULT_RTL_VALUE)
+        }
+    }
+
+    private fun updateCustomStringCheckBox() {
+        customTranslationCheckBox.isChecked = sharedPreference
+            .getBoolean(LANGUAGE_CUSTOM_TRANSLATION_ENABLE, DEFAULT_CUSTOM_TRANSLATION_VALUE)
+    }
+
+    private fun getDeviceLanguage(): String {
+        return Locale.getDefault().displayLanguage
+    }
+
+    private fun setLanguageInSharedPrefForFirstTime() {
+        val deviceLocaleLanguage = getDeviceLanguage()
+        if (isFirstRun()) {
+            for (language in supportedLanguages) {
+                if (language == deviceLocaleLanguage) {
+                    setLanguageValueInSharedPref(language)
+                    break
+                }
+            }
+            if (isFirstRun()) setLanguageValueInSharedPref(DEFAULT_LANGUAGE_VALUE)
+        }
+    }
+
+    private fun setLanguageInAdapter() {
+
+        autoCompleteTextView.setText(
+            sharedPreference.getString(
+                LANGUAGE_ADAPTER_VALUE_SHARED_PREF_KEY,
+                DEFAULT_LANGUAGE_VALUE
+            ),
+            true
+        )
+        languageArrayAdapter.filter.filter(null)
+    }
+
+    private fun isFirstRun(): Boolean {
```

Comment:
With kotlin it's more succinct to use getter syntax, i.e.

val isFirstRun get() = condition

Can leave both the type out (inference) and makes it easier to use in code (i.e. isFirstRun instead of isFirstRun())